x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -78,7 +79,13 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public Object extract(SearchHit hit) {
         Object value = null;
-        DocumentField field = hit.field(fieldName);
+        DocumentField field = null;
+        if (hitName != null) {
+            // a nested field value is grouped under the nested parent name (ie dep.dep_name lives under "dep":[{dep_name:value}])
+            field = hit.field(hitName);
+        } else {
+            field = hit.field(fieldName);
```

Comment:
To make sure, the fieldName is in this case a "doc-top-level" field, right?

Reply:
`fieldName` is the full field name (including the path): `dep.dep_name` for example.